torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -669,21 +670,25 @@ void ProcessGroupNCCL::workCleanupLoop() {
           std::chrono::milliseconds(kWorkCleanupThreadSleepMillis),
           [&]() -> bool { return terminateProcessGroup_.load(); });
 
-      for (auto it = workMetaList_.begin(); it != workMetaList_.end();
-           /* no increment*/) {
-        auto& work = *it;
-        if (work.isCompleted()) {
-          // Handle Exceptions on failed GPU operations and remove completed
-          // workNCCL objects from work vector.
-          if (!terminateProcessGroup_.load()) {
-            work.handleNCCLGuard();
+      if (terminateProcessGroup_.load()) {
```

Comment:
I'm not sure if these conditionals help since the variable could be set to true right after you do this check (same goes for line 682). Is a better fix joining the workCleanupThread before aborting the communicators in ProcessGroupNCCL destructor?

Reply:
Ah ok, so the root cause of those false positive error logs are due to us aborting the communicators during destruction. Yes, this seems like a better fix